lib/ts/recipe/session/api/implementation.ts

```diff
@@ -14,34 +15,27 @@ export default class APIImplementation implements APIInterface {
     }: {
         verifySessionOptions: VerifySessionOptions | undefined;
         options: APIOptions;
-    }): Promise<void> => {
-        try {
-            let method = normaliseHttpMethod(options.req.method);
-            if (method === "options" || method === "trace") {
-                return options.next();
-            }
+    }): Promise<SessionContainerInterface | undefined> => {
+        let method = normaliseHttpMethod(options.req.getMethod());
+        if (method === "options" || method === "trace") {
+            return undefined;
+        }
 
-            let incomingPath = new NormalisedURLPath(
-                options.req.originalUrl === undefined ? options.req.url : options.req.originalUrl
-            );
+        let incomingPath = new NormalisedURLPath(options.req.getOriginalURL());
 
-            let refreshTokenPath = options.config.refreshTokenPath;
+        let refreshTokenPath = options.config.refreshTokenPath;
 
-            if (incomingPath.equals(refreshTokenPath) && method === "post") {
-                (options.req as SessionRequest).session = await options.recipeImplementation.refreshSession({
-                    req: options.req,
-                    res: options.res,
-                });
-            } else {
-                (options.req as SessionRequest).session = await options.recipeImplementation.getSession({
-                    req: options.req,
-                    res: options.res,
-                    options: verifySessionOptions,
-                });
-            }
-            return options.next();
-        } catch (err) {
-            options.next(err);
+        if (incomingPath.equals(refreshTokenPath) && method === "post") {
+            return await options.recipeImplementation.refreshSession({
```

Comment:
How does returning the `session` object attach it to the req object?

Reply:
this is handled by the middleware wrapper defined by each wrapper

Reply:
understood.